dataprofiler/data_readers/json_data.py

```diff
@@ -236,36 +237,38 @@ def _get_data_as_flattened_dataframe(self, json_lines):
 
         return data
 
-    def _load_data_from_str(self, data_as_str: str) -> List:
+    def _load_data_from_str(self, data_as_str: str) -> JSONType:
         """
         Load the data from a string.
 
         :param data_as_str: data in string format.
         :type data_as_str: str
-        :return: dict
+        :return: JSONType
         """
+        data: JSONType
         try:
             data = json.loads(data_as_str)
         except json.JSONDecodeError:
-            data = data_utils.data_generator(data_as_str.splitlines())
+            data_generator = data_utils.data_generator(data_as_str.splitlines())
             data = data_utils.read_json(
-                data_generator=data,
+                data_generator=data_generator,
                 selected_columns=self.selected_keys,
```

Comment:
was this raised by `mypy`? Pretty sure the idea was to overwrite variable names to keep space to a minimum, but could be wrong here.

Reply:
When I explicitly gave a type for `data`, mypy had a issue with this line because `data_generator` returns a generator type. We eventually return `data` so I didn't want to make its type more confusing so I used a different variable name.